### ISSUE TITLE
`volatile` keyword unsupported on structure declaration

### ISSUE THREAD

**Issue by andrewvaughanj** [ltcmelo/psychec#16]
**If eligible for the bounty, I wish this to be donated to a local animal charity of the maintainer's choice**

#### GCC version:
```bash
gcc (SUSE Linux) 10.2.1 20201202 [revision e563687cf9d3d1278f45aaebd03e0f66531076c9]
```

#### clang version:
```bash
clang version 11.0.1
```

#### Output of `gcc -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `clang -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `cnip $FNAME`
```bash
volatile_typedef_struct.c:4:2 error: expected <identifier> or `(' starting direct-declarator, got `volatile'
} volatile st;
  ^

(return code 0)
```

#### Content of file
```bash
struct {
  int m;
} volatile st;
```

#### `psychec` version
```bash
commit 6d7b8249e0f28af105112b217cb33a24f666e148
Author: Leandro T. C. Melo <LTCMELO@GMAIL.COM>
Date:   Tue Feb 23 07:47:25 2021 -0300

    link expression-parsing table in Doxygen documentation
```

**Comment by ltcmelo** [comment]
Hm… I'd guess that the same issue will happen with other type qualifiers. Thanks @andrewvaughanj !

**Comment by ltcmelo** [comment]
Fixed with 7141e1dfb12db6edc736127b1112ee677852b5f5.
